### ISSUE TITLE
[Distributed Index] sync_gateway panic when starting without feed_type.num_shards

### ISSUE THREAD

**Issue by sethrosetter** [couchbase/sync_gateway#1298]
Using
"Couchbase Sync Gateway/0.0.1(360;b85b1c8)"

This configuration launches as expected

```
{
    "interface":":4984",
    "adminInterface": "0.0.0.0:4985",
    "maxIncomingConnections": 0,
    "maxCouchbaseConnections": 16,
    "maxFileDescriptors": 90000,
    "slowServerCallWarningThreshold": 500,
    "compressResponses": true,
    "log":["*"],
    "verbose":"true",
    "cluster_config": {
        "server":"http://{{ couchbase_server_primary_node }}:8091",
        "data_dir":".",
        "bucket":"data-bucket"
    },
    "databases": {
        "db":{
            "feed_type":"DCPSHARD",
            "feed_params":{
                "num_shards":64
            },
            "server":"http://{{ couchbase_server_primary_node }}:8091",
            "bucket":"data-bucket",
            "channel_index":{
                "server":"http://{{ couchbase_server_primary_node }}:8091",
                "bucket":"index-bucket",
                "writer":{{ is_index_writer }}
            }
        },
        "db2":{
            "feed_type":"DCPSHARD",
            "feed_params":{
                "num_shards":64
            },
            "server":"http://{{ couchbase_server_primary_node }}:8091",
            "bucket":"data-bucket-2",
            "channel_index":{
                "server":"http://{{ couchbase_server_primary_node }}:8091",
                "bucket":"index-bucket-2",
                "writer":{{ is_index_writer }}
            }
        }
    }
}
```

This generated a panic with the following error (https://gist.github.com/sethrosetter/d81e6dbf3e71da7fd99a)

Expected (Ideal): sync_gateway would launch with a sensible num_shard default
Expected (min): sync_gateway would gracefully handle the missing property

```
{
    "interface":":4984",
    "adminInterface": "0.0.0.0:4985",
    "maxIncomingConnections": 0,
    "maxCouchbaseConnections": 16,
    "maxFileDescriptors": 90000,
    "slowServerCallWarningThreshold": 500,
    "compressResponses": true,
    "log":["*"],
    "verbose":"true",
    "cluster_config": {
        "server":"http://{{ couchbase_server_primary_node }}:8091",
        "data_dir":".",
        "bucket":"data-bucket"
    },
    "databases": {
        "db":{
            "feed_type":"DCPSHARD",
            "server":"http://{{ couchbase_server_primary_node }}:8091",
            "bucket":"data-bucket",
            "channel_index":{
                "server":"http://{{ couchbase_server_primary_node }}:8091",
                "bucket":"index-bucket",
                "writer":{{ is_index_writer }}
            }
        },
        "db2":{
            "feed_type":"DCPSHARD",
            "server":"http://{{ couchbase_server_primary_node }}:8091",
            "bucket":"data-bucket-2",
            "channel_index":{
                "server":"http://{{ couchbase_server_primary_node }}:8091",
                "bucket":"index-bucket-2",
                "writer":{{ is_index_writer }}
            }
        }
    }
}
```

**Comment by adamcfraser** [comment]
Will be addressed by #1297.

**Comment by adamcfraser** [comment]
Decision has been made to set default value to 64, and have the field non-mandatory.
